uefi-mkconfig

```diff
@@ -53,156 +25,86 @@ add_uefi_entries () (
 		local efi_file_path
 		efi_file_path="${efi_file//$partition_mount}"
 
-		# Add entry if it doesn't exist
-		if ! check_if_uefi_entry_exists; then
-		
-			# Find first free hex address larger than or equal to 0100 for new UEFI boot entry
-			local bootnum
-			bootnum=256 # 256 is decimal value of 0100
-			while [[ "$(efibootmgr -u)" == *"Boot$(printf %04X $bootnum)"* ]]; do
-				bootnum=$((bootnum + 1))
-				
-				# Die if script exceeds managed range 0100-0200
-				if [[ $bootnum -gt 512 ]]; then
-					die "All IDs, within managed range by uefi-mkconfig, are full!"
-				fi
-
-			done
+		# Find first free hex address larger than or equal to 0100 for new UEFI boot entry
+		local bootnum
+		bootnum=256 # 256 is decimal value of 0100
+		while [[ "$(efibootmgr -u)" == *"Boot$(printf %04X $bootnum)"* ]]; do
+			bootnum=$((bootnum + 1))
 			
-			# Convert chosen entry ID into hex
-			bootnum="$(printf %04X $bootnum)"
-
-			# Get kernel version
-			## Remove everything before /
-			local kernel_version
-			kernel_version="${efi_file_path//*\//}"
-			## Remove everything before first - to remove any prefixes
-			kernel_version="${kernel_version/${kernel_version%%-*}-/}"
-			## Remove .efi suffix if it exists
-			if [[ "$kernel_version" == *".efi"* ]]; then
-				kernel_version="${kernel_version/\.${kernel_version##*.}}"
+			# Die if script exceeds managed range 0100-0200
+			if [[ $bootnum -gt 512 ]]; then
+				die "All IDs, within managed range by uefi-mkconfig, are full!"
```

Comment:
Since we remove all entries in our range, this situation should never occur right?

Reply:
If you have hundreds of kernels for some reason, it can still happen. It is improbable but since it's there, why not leave it.

Reply:
Good point